mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Controller/OverviewStatisticsController.php

```diff
@@ -0,0 +1,137 @@
+
+    /** @var Helper */
+    private $wooCommerceHelper;
+
+  public function __construct(
+    NewslettersRepository $newslettersRepository,
+    NewsletterStatisticsRepository $newsletterStatisticsRepository,
+    Helper $wooCommerceHelper
+  ) {
+      $this->newslettersRepository = $newslettersRepository;
+      $this->newsletterStatisticsRepository = $newsletterStatisticsRepository;
+      $this->wooCommerceHelper = $wooCommerceHelper;
+  }
+
+  public function getStatisticsForAutomation(Automation $automation, Query $query): array {
+      $emails = $this->getEmailsFromAutomation($automation);
+      $formattedEmptyRevenue = $this->wooCommerceHelper->getRawPrice(
+        0,
+        [
+          'currency' => $this->wooCommerceHelper->getWoocommerceCurrency(),
+        ]
+      );
+      $data = [
+        'total' => ['current' => 0, 'previous' => 0],
+        'opened' => ['current' => 0, 'previous' => 0],
+        'clicked' => ['current' => 0, 'previous' => 0],
+        'orders' => ['current' => 0, 'previous' => 0],
+        'revenue' => ['current' => 0, 'previous' => 0],
+        'revenue_formatted' => [
+          'current' => $formattedEmptyRevenue,
+          'previous' => $formattedEmptyRevenue,
+        ],
+      ];
+      if (!$emails) {
+        return $data;
+      }
+
+      $requiredData = [
+        'totals',
+        StatisticsClickEntity::class,
+        StatisticsOpenEntity::class,
+        WooCommerceRevenue::class,
+      ];
+
+      $currentStatistics = $this->newsletterStatisticsRepository->getBatchStatistics(
+        $emails,
+        $query->getPrimaryAfter(),
+        $query->getPrimaryBefore(),
+        $requiredData
+      );
+    foreach ($currentStatistics as $statistic) {
+      $data['total']['current'] += $statistic->getTotalSentCount();
+      $data['opened']['current'] += $statistic->getOpenCount();
+      $data['clicked']['current'] += $statistic->getClickCount();
+      $data['orders']['current'] += $statistic->getWooCommerceRevenue() ? $statistic->getWooCommerceRevenue()->getOrdersCount() : 0;
+      $data['revenue']['current'] += $statistic->getWooCommerceRevenue() ? $statistic->getWooCommerceRevenue()->getValue() : 0;
+    }
+
+      $previousStatistics = $this->newsletterStatisticsRepository->getBatchStatistics(
+        $emails,
+        $query->getSecondaryAfter(),
+        $query->getSecondaryBefore(),
+        $requiredData
+      );
+
+    foreach ($previousStatistics as $statistic) {
+      $data['total']['previous'] += $statistic->getTotalSentCount();
+      $data['opened']['previous'] += $statistic->getOpenCount();
+      $data['clicked']['previous'] += $statistic->getClickCount();
+      $data['orders']['previous'] += $statistic->getWooCommerceRevenue() ? $statistic->getWooCommerceRevenue()->getOrdersCount() : 0;
+      $data['revenue']['previous'] += $statistic->getWooCommerceRevenue() ? $statistic->getWooCommerceRevenue()->getValue() : 0;
+    }
+
```

Comment:
This is a so-called "zigzag" indentation 😂

Reply:
:face_exhaling: done